.travis/deploy_s3.sh

```diff
@@ -0,0 +1,246 @@
+patch -p1 -i .travis/gh-5114-update-repo-sh-fix-unbound-var-access.patch
+
+# Decrypt a GPG key.
+gpg_key_file=".travis/deploy_${configuration}_s3_gpg_private_key.asc"
+openssl aes-256-cbc -K "${DEPLOY_S3_GPG_KEY_FILE_KEY}"           \
+    -iv "${DEPLOY_S3_GPG_KEY_FILE_IV}" -in "${gpg_key_file}.enc" \
+    -out "${gpg_key_file}" -d
+
+# Import GPG key for signing repository files.
+gpg --import --batch "${gpg_key_file}"
+
+# Extract GPG key id for signing repository files.
+#
+# This way works for both GnuPG 1 and GnuPG 2. The alternative
+# would be using '--import-options show-only', but it is available
+# only in GnuPG 2. See https://unix.stackexchange.com/a/468889
+mkdir -m 0700 temp-gpg-home
+gpg --homedir temp-gpg-home --import --batch "${gpg_key_file}"
+export GPG_SIGN_KEY="$(gpg --homedir temp-gpg-home --list-secret-keys \
+    --with-colons | grep ^sec: | cut -d: -f5)"
+rm -rf temp-gpg-home
+
+# Use SHA256 hashing algorithm for files signing.
+#
+# `apt-get update` gives a warning when InRelease file signature
+# is calculated with SHA1. We should configure GnuPG (which is
+# used by reprepro, which is used by update_repo.sh) to sign using
+# SHA265.
+#
+# https://askubuntu.com/a/819868
+mkdir -p ~/.gnupg
+echo 'digest-algo sha256' >> ~/.gnupg/gpg.conf
+
+# Setup arguments that are common for all repositories
+# (1.10, 2.1, ...).
+update_repo_args="--os=${OS} --distribution=${DIST}"
+
+# ${PRODUCT} value may affect location of *.deb, *.rpm and related
+# files relative to a base repository URL. We can provide it or
+# miss: the script will generate correct repository metainfo
+# anyway.
+#
+# However providing meaningful value for this option enables
+# grouping of related set of packages into a subdirectory named as
+# ${PRODUCT} (only for Deb repositories at moment of writing
+# this).
+#
+# It is enabled here for consistency with locations of other Deb
+# packages in our repositories, but in fact it is the internal
+# detail, which does not lead to any change in the user
+# experience.
+
+# Guess PRODUCT value in the similar way as packpack does.
+#
+# Guess from Debian package
+[ -z "${PRODUCT:-}" ] && PRODUCT="$(grep Source: debian/control \
+    2>/dev/null | awk '{ print $2; }' || true)"
+# Guess from Debian package (control file template)
+[ -z "${PRODUCT:-}" ] && PRODUCT="$(grep Source: debian/control.in \
+    2>/dev/null | awk '{ print $2; }' || true)"
+# Guess from RPM package
+[ -z "${PRODUCT:-}" ] && PRODUCT="$(grep Name: rpm/*.spec \
+    2>/dev/null | awk '{ print $2; }' || true)"
+# Guess from git repository name
+[ -z "${PRODUCT:-}" ] && PRODUCT="$(git config --get remote.origin.url | \
+    sed -e 's/.*\///' -e 's/.git$//' || true)"
+# Guess from directory name
+[ -z "${PRODUCT:-}" ] && PRODUCT="$(basename "$(pwd)" || true)"
+
+# Add --product option if there is ${PRODUCT} value. Otherwise it
+# will fall back to default 'tarantool'.
+if [ -n "${PRODUCT:-}" ]; then
```

Comment:
It is not necessary here, because the guessing block above will set the variable (possibly to the empty value). However I think it is good style when `set -u` is used. This way this block will work correctly as with the guessing block above as well as without it.

For `DEPLOY_PRODUCTION_S3_ACCESS_KEY_ID` and other variables below: yep, it is necessary, because it may be unset when only staging configuration is provided.

Reply:
I would leave it as is.